### DISCUSSION TITLE
Prisma seed using csv data

### DISCUSSION THREAD

**Discussion by izaacdb** [prisma/prisma#10260] [Q&A]
I've been running into some data-size problems where I need to import huge amounts of seed data. Node runs out of memory with JSON, and is very slow, so I've been importing directly via CSV to Postgres.

The problem is that I am unsure how to run a CSV import as part of the `seed.ts` file. Some of the data is required before the seed, and some can only be created afterwards.

Does anyone have any ideas on how to get around this? I'm guessing I could just separate the seed functionality away from the Prisma CLI but then I would lose the `prisma reset` functionality which is very useful for me. Maybe I could call a Postgres import at the start of the seed file somehow?

**Comment by austincrim** [comment]
Hi @izaacdb 👋, great question!

If you are doing a CSV import via SQL (with a `COPY` query), you could wire that up in a seed file using `prisma.$executeRaw`.

Otherwise, since it sounds like your requirements are pretty complex, you would have to write some custom logic in your seed file to read and parse the CSV file before writing it using Prisma.

It's important to note that the `seed` command that you specify in your `package.json` can actually be any terminal command. It is frequently used to run a Node script, but you could theoretically run a plain .SQL file or [any other language](https://www.prisma.io/docs/guides/database/seed-database#seeding-your-database-via-any-language-with-a-bash-script) for that matter.

If you are imagining a more tight integration with `prisma db seed`, it would be awesome if you could fill out a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=)!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
